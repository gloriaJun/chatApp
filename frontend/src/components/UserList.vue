<template>
  <v-list>
    <v-subheader
      v-if="title"
    >
      {{ title }}
    </v-subheader>

    <v-list-tile
      v-for="item in items"
      :key="item.id"
      avatar
      subheader
      @click='onClickList(item)'
    >
      <v-list-tile-avatar
        color="indigo"
      >
          <span
            class="white--text headline"
          >
            {{ item.username.charAt(0).toUpperCase() }}
          </span>
      </v-list-tile-avatar>

      <v-list-tile-content>
        <v-list-tile-title v-html="item.username"></v-list-tile-title>
      </v-list-tile-content>
    </v-list-tile>
  </v-list>
</template>

<script>
export default {
  name: 'UserList',
  props: {
    title: {
      type: String,
      default: '',
    },
    items: {
      type: [Object, Array],
      default: () => [],
    },
  },
  methods: {
    onClickList(item) {
      this.$emit('click', item);
    },
  },
};
</script>
