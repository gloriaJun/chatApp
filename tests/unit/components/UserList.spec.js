import {
  mount,
} from '@vue/test-utils';
import '../test-setup';

import UserList from '@/components/UserList.vue';

describe('UserList.vue', () => {
  it('snapshot check', () => {
    const wrapper = mount(UserList);

    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it('snapshot check with items', () => {
    const items = [{
      socketId: 'afefeffa',
      username: 'user01',
    }];
    const wrapper = mount(UserList, {
      propsData: { items },
    });

    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it('set title', () => {
    const items = [{
      socketId: 'afefeffa',
      username: 'user01',
    }];
    const title = 'subTitle';
    const wrapper = mount(UserList, {
      propsData: {
        items,
        title,
      },
    });

    expect(wrapper.find('.v-subheader').text()).toBe(title);
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it('click room', () => {
    const items = [{
      socketId: 'afefeffa',
      username: 'user01',
    }];
    const wrapper = mount(UserList, {
      propsData: { items },
    });

    wrapper.find('.v-list__tile').trigger('click');
    expect(wrapper.emitted().click[0]).toEqual([items[0]]);
  });
});
