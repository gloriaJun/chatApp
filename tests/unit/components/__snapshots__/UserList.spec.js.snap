// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList.vue set title 1`] = `
<div
  class="v-list theme--light"
  role="list"
>
  <div
    class="v-subheader theme--light"
  >
    
    subTitle
  
  </div>
   
  <div
    role="listitem"
  >
    <a
      class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light"
      subheader=""
    >
      <div
        class="v-list__tile__avatar"
      >
        <div
          class="v-avatar indigo"
          style="height: 40px; width: 40px;"
        >
          <span
            class="white--text headline"
          >
            
          U
        
          </span>
        </div>
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          user01
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`UserList.vue snapshot check 1`] = `
<div
  class="v-list theme--light"
  role="list"
/>
`;

exports[`UserList.vue snapshot check with items 1`] = `
<div
  class="v-list theme--light"
  role="list"
>
  <!---->
   
  <div
    role="listitem"
  >
    <a
      class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light"
      subheader=""
    >
      <div
        class="v-list__tile__avatar"
      >
        <div
          class="v-avatar indigo"
          style="height: 40px; width: 40px;"
        >
          <span
            class="white--text headline"
          >
            
          U
        
          </span>
        </div>
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          user01
        </div>
      </div>
    </a>
  </div>
</div>
`;
